content/en/blog/memory_management/index.md

```diff
@@ -0,0 +1,141 @@
+hot-plug/unplug, relying on the `virtio-mem` driver in the guest to manage small
+chunks of memory. Unless you have strong reasons not to use `virtio-mem` (guest
+kernel not recent enough for instance), this should be the standard way of
+dynamically managing a VM's memory.
+
+### Guest driven
+
+Now that we've covered how the host can manage the guest memory allocated to a
+VM, let's look at the way the guest can notify the host for accurate memory
+management.
+
+#### Free page reporting
+
+In general, when a VM is created with a certain amount of memory, only a small
+portion of what is available is actually consumed by the VM after boot. But as
+soon as one or multiple workloads run inside the guest, the memory starts being
+paged. At this point, let's say 90% of the VM's memory have been allocated, the
+guest might start freeing some pages after the workloads have been stopped. This
+will have no impact on the amount of memory consumed from a host perspective, as
+it has no way to know which pages the guest might have freed.
+
+When looking for a way to overcommit memory, meaning when running multiple VMs
+and the sum of all guest RAMs is higher than what the physical RAM can offer,
+being able to free pages based on guest's input is a must. And that's what the
+feature `free page reporting` has been added for in the `virtio-balloon`
+specification.
+
+This feature requires a `virtio-balloon` device to be created, but it doesn't
+require any balloon because inflating or deflating operations are not part of
+the mechanism. When enabled, the guest will have a way of notifying the VMM
+about set of pages being freed. Based on this information, the VMM will advise
+the host that these pages are no longer in use, which effectively gives some
+memory back to the host.
+
+One small drawback of this feature is that depending on how frequently the guest
+reports back to the host, the VMM will have more work and be slightly less
+efficient. And that's because the slight drop in performance that could be
+observed that we didn't enable this feature by default on Cloud Hypervisor.
+
+#### Deflate on OOM
+
+When running critical workloads that must not fail, particularly because of
+memory pressure in the guest, the feature `deflate on OOM` can be very
+convenient as it will give the guest the power to deflate the `balloon` when
+an Out Of Memory (OOM) event happens.
+
+Of course this means the `balloon` must have been previously inflated, otherwise
+guest will have nothing to deflate and no way to release the memory pressure.
+
+### References
+
+If you want to use these features with Cloud Hypervisor, please refer to the
+following documentation:
+- [Memory ballooning](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/balloon.md)
+- [Memory hot-plug](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/memory.md)
+- [Guest features](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/balloon.md)
```

Comment:
Is it intentional that this is the same as the balloon?

Reply:
It is intentional. Do you think I should change that?